src/NServiceBus.Core/Persistence/Learning/SagaPersister/LearningSagaIdGenerator.cs

```diff
@@ -0,0 +1,25 @@
+namespace NServiceBus
+{
+    using System;
+    using Sagas;
+
+    class LearningSagaIdGenerator : ISagaIdGenerator
+    {
+        public Guid Generate(SagaIdGeneratorContext context)
+        {
+            if (context.CorrelationProperty == SagaCorrelationProperty.None)
+            {
+                throw new NotSupportedException("The development saga persister doesn't support custom saga finders.");
```

Comment:
Should we move this to the init phase of the persister and prevent startup if customer finders are use?